compaction/compaction_manager.hh

```diff
@@ -75,8 +91,15 @@ private:
         bool compaction_running = false;
         utils::UUID output_run_identifier;
         sstables::compaction_data compaction_data;
+        compaction_state& compaction_state;
```

Comment:
It's interesting to keep a ref to state in task, so we won't have to look it up multiple times.

Reply:
it's more than about saving a lookup, it's for the ability to work with it (e.g. unlock) while it's detached from the `_compaction_state` map.

Reply:
makes sense. missed that we're now detaching state from map in advance. thanks for the explanation!